cns/restserver/ipam.go

```diff
@@ -661,8 +675,15 @@ func (service *HTTPRestService) GetExistingIPConfig(podInfo cns.PodInfo) ([]cns.
 
 // Assigns a pod with all IPs desired
 func (service *HTTPRestService) AssignDesiredIPConfigs(podInfo cns.PodInfo, desiredIPAddresses []string) ([]cns.PodIpInfo, error) {
+	// Sets the number of desired IPs equal to the number of NCs so that we can get one IP per NC
 	numDesiredIPAddresses := len(desiredIPAddresses)
+	// Creates a slice of PodIpInfo with the size as number of NCs to hold the result for assigned IP configs
 	podIPInfo := make([]cns.PodIpInfo, numDesiredIPAddresses)
+	// if there are no NCs on the NNC there will be no IPs in the pool so return error
+	if numDesiredIPAddresses == 0 {
+		//nolint:goerr113 // return error
```

Comment:
This linter bypass is inappropriate here. The error returned below should be defined like:

```go
var errNoPoolIPs = errors.New("no NCs found on the NNC so no IPs are in the pool")
```

and consequently used like:

```go
return nil, errNoPoolIPs
```

**Note also the `nil`, instead of `podIPInfo`**. In cases where an error is returned, it should be a nil or zero value being returned first.

This also applies to all of the other instances of `//nolint:goerr113 // return error` for this func.

Reply:
Added a `ErrNoNCs` error variable and check for it in the UTs as well.